test/github/client/github-client.test.ts

```diff
@@ -272,6 +278,53 @@ describe("GitHub Client", () => {
 		expect(error).toBeInstanceOf(GithubClientError)
 		expect(error.status).toBe(404)
 
-		verifyMetricsSent('/repos/:owner/:repo/pulls', "404");
+		verifyMetricsSent("/repos/:owner/:repo/pulls", "404");
 	});
+
+	/**
+	 * One test against a non-cloud GitHub URL to prove that the client will be working against an on-premise
+	 * GHE installation.
+	 */
+	it("works with a non-cloud installation", async () => {
+		const owner = "owner";
+		const repo = "repo";
+		const pageSize = 5;
+		const page = 1;
+
+		givenGitHubReturnsInstallationToken("installation token", gheNock);
+		givenGitHubReturnsPullrequests(
+			owner,
+			repo,
+			pageSize,
+			page,
+			"installation token",
+			gheNock
+		);
+
+		const appTokenHolder = new AppTokenHolder((installationId: InstallationId) => {
+			switch (installationId.githubBaseUrl) {
+				case "https://api.github.com":
+					return "cloud private key";
+				case "http://github.mydomain.com":
+					return "GHE private key";
+				default:
+					throw new Error("unknown github instance!");
+			}
+		});
+
+		const client = new GitHubClient(
+			new InstallationId("http://github.mydomain.com", 4711, githubInstallationId),
```

Comment:
@rachellerathbone this is how we will need to instantiate a `GitHubClient` in the code that's handling a webhook from GHE. All values in the `InstallationId` (URL, appId, installationId) we can hard-code for the spike.